ci-operator/step-registry/cucushift/installer/check/aws/custom-sg/cucushift-installer-check-aws-custom-sg-commands.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+  export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+  echo "No KUBECONFIG found, exit now"
+  exit 1
+fi
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+if [[ ! -f ${SHARED_DIR}/security_groups_ids ]]; then
+    echo "No custom SG was created, exit now."
+    exit 1
+fi
+
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+ret=0
+
+if [[ "${ENABLE_CUSTOM_SG_DEFAULT_MACHINE}" == "true" || "${ENABLE_CUSTOM_SG_CONTROL_PLANE}" == "true" ]]; then
+
+    echo "Custom security group enabled on control planes, check the SG attched to control plane"
+    # get SG from one of the masters
+    control_plane_sgs=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:Name,Values=${INFRA_ID}-master*" | jq -r '.Reservations[0].Instances[].NetworkInterfaces[].Groups[].GroupId')
+
+    for sg_id in $(cat ${SHARED_DIR}/security_groups_ids); do
+        if [[ ${control_plane_sgs} =~ ${sg_id} ]]; then
```

Comment:
@gpei  If only one sg id presents in `security_groups_ids`, we can remove `for` block I think.
If more than one IDs, it looks like the logic is not correct.

Reply:
@yunjiang29 The output of ${control_plane_sgs} is a long string with all the SGs based on my test 
```
[root@preserve-gpei-worker k_files]# control_plane_sgs=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:Name,Values=${INFRA_ID}-master*" | jq -r '.Reservations[0].Instances[].NetworkInterfaces[].Groups[].GroupId')
[root@preserve-gpei-worker k_files]# echo $control_plane_sgs
sg-08a8a067ade21aabd sg-0da9df5e84e968c5f sg-002ea59ec099c7adc sg-0b06bc19eb7b28372
```
so I think we can test if each SG is attached in this way directly?